### DISCUSSION TITLE
adding yin segmentation to CaMu

### DISCUSSION THREAD

**Discussion by diemoschwarz** [Ircam-RnD/catart-mubu#84] [Ideas]
On 23/04/2023 10:54, Christopher Trapani wrote:
> Hi guys — I am trying to build a camu.process.yin patch which will apply the yin 
> markers processing that we developed for our Seg-Lab. Far better for grains with 
> good frequency estimation.
> 
> I'm in need of a little help / collaboration, though. The mubu.process modules 
> define the markers well, but could one of you give me a hand or a hint about the 
> next step — namely, writing a mubu.process that will reanalyze the audio 
> according to the markers generate by yin (or resegment the raw yin data) for a 
> table of familiar descr values ?
> 
> I've attached a prototype here, integrated into tutorial 5. Ultimately we could 
> add this (and maybe other segmentation methods!) to the package...
> 
> Thanks — C

**Comment by einbond** [comment]
Yes agreed! Could this relate to the more general issue of applying external markers to mubu.process? #82

**Comment by diemoschwarz** [comment]
Hi guys, segmentation has advanced a lot in Mubu!  There is an improved example of yin segmentation in Matt's 4th [mubu tutorial](https://forum.ircam.fr/projects/detail/mubu-tutorials-material/).
I'll add a quick mubu.process.yin patch, but, indeed, we should move to process2 which allows external and hand-edited segmentation.

**Reply by christophertrapani** [reply]
So, just to understand: that would be a feature request, to create a new process2 object? And it's not possible with (some combination of) current MuBu objects?

Meanwhile, I am watching the tutorial to see if there are lessons we can apply to CaMu...

**Comment by einbond** [comment]
Sorry my mind was rusty on our previous process2 discussions and I now see this is already under issue #48 . I'll close this and we can continue there!

**Comment by diemoschwarz** [comment]
yin seg. discussion continues on issue #86
